### PR TITLE
Added Depth sampling modes to the scene depth node

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR
Add a mode in the Scene Depth node to choose between linear01, raw or eye depth.

@Kink3d  This is a follow up of the conversation about depth sampling on slack.

---
### Testing status
**Manual Tests**:  simple test scene

**Automated Tests**:  none

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**:  Low

**Halo Effect**:  Low

### CODE REVIEW

File: com.unity.shadergraph/Documentation~/Scene-Depth-Node.md [hunk 1]
```diff
@@ -21,12 +21,19 @@ NOTE: This [Node](Node.md) can only be used in the **Fragment** [Shader Stage](S
 | UV     | Input | Vector 4 | Screen Position | Normalized screen coordinates |
 | Out | Output      |    Vector 1 | None | Output value |
 
+## Depth Sampling modes
+| Name     | Description                        |
+|----------|------------------------------------|
+| Linear01 | linear depth value between 0 and 1 |
+| Raw      | raw depth value                    |
+| Eye      | depth converted to world space units |
```
Comment (sebastienlagarde) [inline com.unity.shadergraph/Documentation~/Scene-Depth-Node.md]: this is to Eye space :)

Comment (alelievr) [reply com.unity.shadergraph/Documentation~/Scene-Depth-Node.md]: fixed

File: com.unity.shadergraph/Documentation~/Scene-Depth-Node.md [hunk 2]
```diff
@@ -21,12 +21,19 @@ NOTE: This [Node](Node.md) can only be used in the **Fragment** [Shader Stage](S
 | UV     | Input | Vector 4 | Screen Position | Normalized screen coordinates |
 | Out | Output      |    Vector 1 | None | Output value |
 
+## Depth Sampling modes
+| Name     | Description                        |
+|----------|------------------------------------|
+| Linear01 | linear depth value between 0 and 1 |
```
Comment (Kink3d) [inline com.unity.shadergraph/Documentation~/Scene-Depth-Node.md]: Capitalise the first letter of each description please

File: com.unity.shadergraph/Editor/Data/Nodes/Input/Scene/SceneDepthNode.cs [hunk 3]
```diff
@@ -12,6 +21,23 @@ sealed class SceneDepthNode : CodeFunctionNode, IMayRequireDepthTexture
         public const int ScreenPositionSlotId = 0;
         public const int OutputSlotId = 1;
 
+        [SerializeField]
+        private DepthSamplingMode m_DepthSamplingMode = DepthSamplingMode.Linear01;
+
+        [EnumControl("Sampling Mode")]
+        public DepthSamplingMode depthSamplibgMode
```
Comment (Kink3d) [inline com.unity.shadergraph/Editor/Data/Nodes/Input/Scene/SceneDepthNode.cs]: depthSamplingMode

**Comment by Katstodian** [comment]
This needs a changelog entry? :)

File: com.unity.shadergraph/Documentation~/Scene-Depth-Node.md (hunk 2 above)
Comment (alelievr) [reply com.unity.shadergraph/Documentation~/Scene-Depth-Node.md]: fixed

File: com.unity.shadergraph/Editor/Data/Nodes/Input/Scene/SceneDepthNode.cs (hunk 3 above)
Comment (alelievr) [reply com.unity.shadergraph/Editor/Data/Nodes/Input/Scene/SceneDepthNode.cs]: fixed

**Pull request merged**